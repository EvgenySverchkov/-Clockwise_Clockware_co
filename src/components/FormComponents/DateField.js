import React from "react";

export default ({min, chngHandler, value, name})=>(
    <input
        type="date"
        name={name}
        min={min}
        className="mr-1"
        onChange={chngHandler}
        value={value}
        required
    />
);