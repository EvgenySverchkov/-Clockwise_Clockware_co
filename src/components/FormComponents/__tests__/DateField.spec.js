import React from 'react';
import {create} from "react-test-renderer";

import DateField from "../DateField";
import ContextComponent from "../../../ContextComponent";

it("<DateField /> test which is used in forms", ()=>{
    const mockProps = {
        min: "2020-09-23", 
        chngHandler: jest.fn(), 
        value: '2020-09-23', 
        name: "Name", 
        max: "2021-09-23"
    }
    const mockContextValue = {
        warningTooltipMsg: "Message",
        isOpenWarningTooltip: false,
        idForTooltip: 1
    }
    const component = create(
        <ContextComponent.Provider value={mockContextValue}>
            <DateField {...mockProps}/>
        </ContextComponent.Provider>
    )
    expect(component.toJSON()).toMatchSnapshot();
})