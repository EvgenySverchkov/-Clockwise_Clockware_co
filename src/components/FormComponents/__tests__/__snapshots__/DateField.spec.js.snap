// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateField /> test which is used in forms 1`] = `
<input
  aria-describedby={null}
  className="mr-1"
  id="date"
  max="2021-09-23"
  min="2020-09-23"
  name="Name"
  onBlur={[Function]}
  onChange={[MockFunction]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  required={true}
  title="Message"
  type="date"
  value="2020-09-23"
/>
`;
